src/components/CompareResults/ResultsMain.tsx

```diff
@@ -92,18 +98,154 @@ function ResultsMain() {
+      .replace(/[\u0300-\u036f]/g, '') // Remove accents
+      .toLowerCase()
+      .replace(/[^a-z0-9]+/g, '-') // Convert non-alphanumeric to hyphen
+      .replace(/^-+|-+$/g, ''); // Trim hyphens from start and end
+    return title;
+  };
+
+  const onComparisonTitleChange = (value: string) => {
+    // setPreviousComparisonTitle(comparisonTitleName);
+    setComparisonTitleName(value);
+
+    if (comparisonTitleName) {
+      const slug = slugifyComparisonTitle(comparisonTitleName);
+      rawSearchParams.set('title', slug);
+    } else {
+      rawSearchParams.delete('title');
+    }
+  };
+
+  const OnComparisonTitleSave = () => {
+    if (comparisonTitleName) {
+      updateRawSearchParams(rawSearchParams);
+      showEditComparisonTitle(!editComparisonTitleInputVisible);
+    }
+    setTitleError(true);
+  };
+
+  //save for next commit
+  // const deSlugify = (slug: string | undefined) => {
+  //   if (!slug) return '';
+  //   return slug
+  //     .replace(/-/g, ' ')
+  //     .replace(/\b\w/g, (char) => char.toUpperCase());
+  // };
+
+  // const deSlugifiedTitle = deSlugify(comparisonTitleName);
+
+  const renderEditSaveCancelBtns = () => {
+    return (
+      <>
+        {editComparisonTitleInputVisible ? (
+          <>
+            <Button
+              name='cancel-title'
+              aria-label='cancel title'
+              className='cancel-btn'
+              variant='text'
+              onClick={handleEditInputToggle}
+            >
+              Cancel
+            </Button>
+            <Button
+              name='save-title'
+              aria-label='save title'
+              className='save-btn'
+              variant='text'
+              onClick={OnComparisonTitleSave}
+            >
+              Save
+            </Button>
+          </>
+        ) : (
+          <Button
+            name='edit-title'
+            aria-label='edit title'
+            startIcon={buttonIcon}
+            className='edit-title-btn'
+            variant='text'
+            onClick={handleEditInputToggle}
+            sx={{ fontSize: '0.75rem' }}
+          >
+            Edit title
+          </Button>
+        )}
+      </>
+    );
+  };
+
+  const buttonIcon = (
+    <img
+      id='edit-title-icon'
+      className='icon icon-edit'
+      src={themeMode === 'light' ? pencil.toString() : pencilDark.toString()}
+      alt='edit-icon'
+    />
+  );
+
+  const titleErrorMessage = titleError ? 'Title cannot be empty' : null;
+
+  const renderResultsTableTitleOrInput = () => {
+    return (
+      <>
+        {editComparisonTitleInputVisible ? (
+          <EditTitleInput
+            titleError={titleErrorMessage}
+            compact={true}
+            onChange={onComparisonTitleChange}
+            value={comparisonTitleName}
+          />
+        ) : (
+          <Grid component='h2' item className={styles.title}>
+            {comparisonTitleName}
+          </Grid>
+        )}
+
+        {renderEditSaveCancelBtns()}
```

Comment:
IMO the save/cancel buttons should go in `EditTitleInput` directly, you'd save a lot of comparisons.

Also if you can make it part of a `<form>`, you'll need to just handle the `submit` event for both the save (button type="submit") and the enter key (automatic), with the added benefit of cleaner semantics.

I'd like to add that it would be good if the "Escape" button would do a "cancel" too. You can use the [`key`](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key) property of the keydown event to check for this.